docs: update README with gardener removal (#3389)
kyma-project/control-plane: components/kyma-metrics-collector/README.md

## Functionality
The basic flow for KMC is as follows:
* KMC workers get a list of runtimes from [Kyma Environment Broker (KEB)](https://github.com/kyma-project/kyma-environment-broker/tree/main).

>  * KMC adds the runtimes to a queue to work through them and re-queues a runtime should an error occur.

Reviewer: ```suggestion
 * KMC adds the runtimes to a queue to work through them. If an error occurs, KMC re-queues the affected runtime.
```

Author: Done

>  * Information on PVCs, SVCs and Nodes is retrieved from SAP Kyma Runtime (SKR).

Reviewer: ```suggestion
 * Information on PVCs, SVCs and Nodes is retrieved from SAP BTP, Kyma runtime (SKR).
```
official product name

Author: Done
